### ISSUE TITLE
USER STORY: Validation and Hyperparameter Optimization

### ISSUE THREAD

**Issue by Shida18719** [Shida18719/heritage-housing-issues#9]
As a **data practitioner**, I want to **validate and find the best hyperparameter for the algorithm**, so that I can **use the best feature that best suit the machine learning pipeline.**

# Acceptance Criteria:

- To access the most relevant features from the pipeline.
- To Evaluate the pipeline on the Train and Test Set, ensuring it meets the business requirements.
